enhancements/20220525-system-logging.md

```diff
@@ -0,0 +1,89 @@
+# System Logging
+
+## Summary
+
+We need to be able to support exporting harvester system logs outside the cluster, to allow for collecting and analyzing those logs to aid in debugging and harvester management. 
+
+### Related Issues
+
+- https://github.com/harvester/harvester/issues/577
+
+## Motivation
+
+### Goals
+
+List the specific goals of the enhancement. How will we know that this has succeeded?
+
+- The user can aggregate harvester logs in a centralized location
+- The user can view the aggregated harvester logs outside the cluster
+
+### Non-goals
+
+- this enhancement does not cover integration into the harvester / rancher UI (but ideally this will eventually be implemented as well)
+
+## Proposal
+
+We will deploy a new `ManagedChart` in the [Harvester Installer](https://github.com/harvester/harvester-installer) to
+install. The `ManagedChart` will deploy a [`ClusterFlow`](https://banzaicloud.com/docs/one-eye/logging-operator/configuration/flow/) to select and 
```

Comment:
`Rancher` has integrated logging, do we have a comparing between it and `ClusterFlow` ? The `Rancher logging` integration process looks similar with `rancher-monitoring`, which is used in Harvester.

https://rancher.com/docs/rancher/v2.5/en/logging/

`cattle-logging-system`  `rancher-logging` and `rancher-logging-crd`.

Reply:
I don't have any formal comparison, but proposed implementation is largely replicating the functionality of rancher logging which utilizes Banzai `ClusterFlows`. As long as we're careful about name spacing both should be able to coexist without issue, and can potentially simplify rancher logging by just adding a new `ClusterOutput` to the Harvester cluster rather than installing a new Helm Chart.

Reply:
the plan is to deploy the Rancher logging chart as a managed chart. This will deploy the ClusterFlow, Flow, ClusterOutput and Output CRDs. In addition to the chart, we will also deploy a ClusterFlow which will aggregate the logs from the harvester system namespaces, along with key OS log files.

This can then be configured to a target endpoint via a new Harvester Setting which will map to a ClusterOuput and correlate with the pre-packaged ClusterFlow.

Reply:
Does Rancher logging chart install the [logging-operator](https://banzaicloud.com/docs/one-eye/logging-operator/install/)? Looks like the operator manages these CRDs.

Reply:
`rancher-logging` chart will deploy `logging-operator`

```
values.yaml

# Default values for logging-operator.
...
image:
  repository: rancher/mirrored-banzaicloud-logging-operator
 ...
```